contracts/AelinPool.sol

```diff
@@ -97,18 +105,31 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         aelinDealLogicAddress = _aelinDealLogicAddress;
         aelinRewardsAddress = _aelinRewardsAddress;
 
-        if (_poolData.allowListAddresses.length > 0 || _poolData.allowListAmounts.length > 0) {
+        address[] memory tmpAllowListAddresses = _poolData.allowListAddresses;
+        uint256[] memory tmpAllowListAmounts = _poolData.allowListAmounts;
+
+        if (tmpAllowListAddresses.length > 0 || tmpAllowListAmounts.length > 0) {
             require(
-                _poolData.allowListAddresses.length == _poolData.allowListAmounts.length,
+                tmpAllowListAddresses.length == tmpAllowListAmounts.length,
                 "allowListAddresses and allowListAmounts arrays should have the same length"
             );
-            for (uint256 i = 0; i < _poolData.allowListAddresses.length; i++) {
-                allowList[_poolData.allowListAddresses[i]] = _poolData.allowListAmounts[i];
-                emit AllowlistAddress(_poolData.allowListAddresses[i], _poolData.allowListAmounts[i]);
+            for (uint256 i = 0; i < tmpAllowListAddresses.length; i++) {
+                allowList[tmpAllowListAddresses[i]] = tmpAllowListAmounts[i];
+                emit AllowlistAddress(tmpAllowListAddresses[i], tmpAllowListAmounts[i]);
             }
             hasAllowList = true;
         }
 
+        NftData[] memory tmpNftData = _poolData.nftData;
+
+        if(tmpNftData.length > 0) {
+            require(tmpNftData.length <= 5, "max 5 collections allowed");
```

Comment:
is there a reason you chose 5 here?

Reply:
during our call, i thought you mentioned a total of 5 collections per pool. should we leave it uncapped?

Reply:
yes, let's leave it uncapped. if they can pay the gas for adding 100 then they should be able to do that as long as it fits in a block